nautobot/dcim/models/device_components.py

```diff
@@ -495,10 +495,118 @@ class BaseInterface(RelationshipModel):
         verbose_name="MTU",
     )
     mode = models.CharField(max_length=50, choices=InterfaceModeChoices, blank=True)
+    parent_interface = models.ForeignKey(
+        to="self",
+        on_delete=models.CASCADE,
+        related_name="child_interfaces",
+        null=True,
+        blank=True,
+        verbose_name="Parent interface",
+    )
+    bridge = models.ForeignKey(
+        to="self",
+        on_delete=models.SET_NULL,
+        related_name="bridged_interfaces",
+        null=True,
+        blank=True,
+        verbose_name="Bridge interface",
+    )
 
     class Meta:
         abstract = True
 
+    def clean(self):
+        super().clean()
+
+        # Virtual interfaces cannot be connected
+        if getattr(self, "type", None) in NONCONNECTABLE_IFACE_TYPES and (
+            self.cable or getattr(self, "circuit_termination", False)
+        ):
+            raise ValidationError(
+                {
+                    "type": "Virtual and wireless interfaces cannot be connected to another interface or circuit. "
+                    "Disconnect the interface or choose a suitable type."
+                }
+            )
```

Comment:
This is very specific to Interface and doesn't apply to VMInterface - IMHO it doesn't belong in `BaseInterface.clean`, it should be in `Interface.clean`.

Reply:
If we collapse Device and Virtual Machine (see: https://github.com/nautobot/nautobot/issues/1178) we will have this all in the same space anyway.

Maybe too early of an optimization?

The `getattr(self, "type", None)` will exclude VMInterface's since they do not contain the type field at this time.

Reply:
Too early of an optimization IMHO.